task1/src/app/courceitem/courceitem.component.ts

```diff
@@ -8,20 +9,20 @@ import { CourceItem } from '../cource-item';
 })
 export class CourceitemComponent implements OnInit {
  
-  @Input() courceItem : CourceItem;
+  @Input() courceItem : ICourceItem;
+  @Output() deleteCourceEvent = new EventEmitter<number>();
+  
   courceId: number;
 
   constructor() { }
 
   ngOnInit() {
-    this.courceId = this.courceItem.Id;
+    if(this.courceItem)
+      this.courceId = this.courceItem.Id;
   }
```

Comment:
ngOinit jest wywolany raz, jezeli chesz aby to przypisanie bylo zaszew aktualne powinienes uzyc OnChanges.